examples/llm/common/web_scraper_module.py

```diff
@@ -0,0 +1,144 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import os
+
+import requests
+import requests_cache
+from bs4 import BeautifulSoup
+
+logging
+
+import logging
+from functools import partial
+
+import mrc
+import mrc.core.operators as ops
+import pandas as pd
+from langchain.text_splitter import RecursiveCharacterTextSplitter
+from pydantic import ValidationError
+
+import cudf
+
+from morpheus.messages import MessageMeta
+from morpheus.modules.schemas.examples.llm.web_scraper_schema import WebScraperSchema
+from morpheus.utils.module_utils import ModuleLoaderFactory
+from morpheus.utils.module_utils import register_module
+
+logger = logging.getLogger(__name__)
+
+WebScraperLoaderFactory = ModuleLoaderFactory("web_scraper", "morpheus_examples_llm", WebScraperSchema)
+
+
+def download_and_split(msg: MessageMeta, text_splitter, link_column, session) -> MessageMeta:
+    """
+Uses the HTTP GET method to download/scrape the links found in the message, splits the scraped data, and stores
+it in the output, excludes output for any links which produce an error.
+"""
+    if (link_column not in msg.get_column_names()):
+        return None
+
+    df = msg.df
```

Comment:
This is going to log a warning
```suggestion
    df = msg.copy_dataframe()
```

or if you don't think there will be lock contention you could do:
```
with meta.mutable_dataframe() as df:
```